deploy/test-environments/fleet_api/src/api/agent_policy_api.py

```diff
@@ -0,0 +1,139 @@
+"""
+This module contains API calls related to the agent policy API.
+"""
+
+from munch import Munch, munchify
+from loguru import logger
+from api.headers import base_headers as headers
+from api.base_call_api import APICallException, perform_api_call
+
+
+def create_agent_policy(cfg: Munch, json_policy: dict) -> str:
+    """This function creates an agent policy
+
+    Args:
+        cfg (Munch): Config object containing authentication data.
+        json_policy (dict): Data for the agent policy to be created.
+
+    Returns:
+        str: The ID of the created agent policy
+
+    Raises:
+        APICallException: If the API call fails or returns a non-200 status code.
+    """
+    # pylint: disable=duplicate-code
```

Comment:
is there really a code duplicate here? can we handle some of the code duplications in this PR?

Reply:
In this case, there is no actual code duplication. On the contrary, I am reusing the `perform_api_call` function across all API calls. The reason for the reported duplicate code error is that the arguments passed to the function have the same body content in all cases.
![Screenshot 2023-06-06 at 14 18 28](https://github.com/elastic/cloudbeat/assets/99176494/b19f733c-ccd2-4525-ad3e-a8126c76ee31)